
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="AIDO.ModelGenerator is a software stack powering the development of an AI-driven Digital Organism by enabling researchers to adapt pretrained models and generate finetuned models for downstream tasks.">
      
      
        <meta name="author" content="GenBio AI">
      
      
        <link rel="canonical" href="https://genbio-ai.github.io/ModelGenerator/api_reference/data/">
      
      
        <link rel="prev" href="../backbones/">
      
      
        <link rel="next" href="../tasks/">
      
      
      <link rel="icon" href="../../assets/images/icon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.44">
    
    
      
        <title>Data - AIDO.ModelGenerator</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.0253249f.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
    
    
      <link rel="stylesheet" href="../../assets/_mkdocstrings.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="lime">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#data" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow md-header--lifted" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="AIDO.ModelGenerator" class="md-header__button md-logo" aria-label="AIDO.ModelGenerator" data-md-component="logo">
      
  <img src="../../assets/images/logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            AIDO.ModelGenerator
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Data
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="lime"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="green" data-md-color-accent="lime"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
          <a href="javascript:void(0)" class="md-search__icon md-icon" title="Share" aria-label="Share" data-clipboard data-clipboard-text="" data-md-component="search-share" tabindex="-1">
            
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.15c-.05.21-.08.43-.08.66 0 1.61 1.31 2.91 2.92 2.91s2.92-1.3 2.92-2.91A2.92 2.92 0 0 0 18 16.08"/></svg>
          </a>
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
        <div class="md-search__suggest" data-md-component="search-suggest"></div>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/genbio-ai/ModelGenerator" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    AIDO.ModelGenerator
  </div>
</a>
      </div>
    
  </nav>
  
    
      
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-grid">
    <ul class="md-tabs__list">
      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../.." class="md-tabs__link">
        
  
    
  
  Introduction

      </a>
    </li>
  

      
        
  
  
  
    <li class="md-tabs__item">
      <a href="../../quick_start/" class="md-tabs__link">
        
  
    
  
  Quick Start

      </a>
    </li>
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../usage/" class="md-tabs__link">
          
  
    
  
  Basic Usage

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../tutorials/kfold_cross_validation/" class="md-tabs__link">
          
  
    
  
  Tutorials

        </a>
      </li>
    
  

      
        
  
  
  
    
    
      <li class="md-tabs__item">
        <a href="../../experiment_design/" class="md-tabs__link">
          
  
    
  
  Experiment Design

        </a>
      </li>
    
  

      
        
  
  
    
  
  
    
    
      <li class="md-tabs__item md-tabs__item--active">
        <a href="../backbones/" class="md-tabs__link">
          
  
    
  
  API Reference

        </a>
      </li>
    
  

      
    </ul>
  </div>
</nav>
    
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="AIDO.ModelGenerator" class="md-nav__button md-logo" aria-label="AIDO.ModelGenerator" data-md-component="logo">
      
  <img src="../../assets/images/logo.png" alt="logo">

    </a>
    AIDO.ModelGenerator
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/genbio-ai/ModelGenerator" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.6.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    AIDO.ModelGenerator
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Introduction
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../quick_start/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Quick Start
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
    
    
      
      
        
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../usage/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Basic Usage
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
      
      
        
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../tutorials/kfold_cross_validation/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Tutorials
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
    
    
      
      
        
      
    
    
      
        
        
      
    
    <li class="md-nav__item md-nav__item--pruned md-nav__item--nested">
      
        
  
  
    <a href="../../experiment_design/" class="md-nav__link">
      
  
  <span class="md-ellipsis">
    Experiment Design
  </span>
  

      
        <span class="md-nav__icon md-icon"></span>
      
    </a>
  

      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
      
        
        
      
      
        
      
    
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="">
            
  
  <span class="md-ellipsis">
    API Reference
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            API Reference
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../backbones/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Backbones
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Data
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Data
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#dna" class="md-nav__link">
    <span class="md-ellipsis">
      DNA
    </span>
  </a>
  
    <nav class="md-nav" aria-label="DNA">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.NTClassification" class="md-nav__link">
    <span class="md-ellipsis">
      NTClassification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.GUEClassification" class="md-nav__link">
    <span class="md-ellipsis">
      GUEClassification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ClinvarRetrieve" class="md-nav__link">
    <span class="md-ellipsis">
      ClinvarRetrieve
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.PromoterExpressionRegression" class="md-nav__link">
    <span class="md-ellipsis">
      PromoterExpressionRegression
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.PromoterExpressionGeneration" class="md-nav__link">
    <span class="md-ellipsis">
      PromoterExpressionGeneration
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DependencyMappingDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      DependencyMappingDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rna" class="md-nav__link">
    <span class="md-ellipsis">
      RNA
    </span>
  </a>
  
    <nav class="md-nav" aria-label="RNA">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TranslationEfficiency" class="md-nav__link">
    <span class="md-ellipsis">
      TranslationEfficiency
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ExpressionLevel" class="md-nav__link">
    <span class="md-ellipsis">
      ExpressionLevel
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TranscriptAbundance" class="md-nav__link">
    <span class="md-ellipsis">
      TranscriptAbundance
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ProteinAbundance" class="md-nav__link">
    <span class="md-ellipsis">
      ProteinAbundance
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.NcrnaFamilyClassification" class="md-nav__link">
    <span class="md-ellipsis">
      NcrnaFamilyClassification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SpliceSitePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      SpliceSitePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ModificationSitePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      ModificationSitePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.RNAMeanRibosomeLoadDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      RNAMeanRibosomeLoadDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#protein" class="md-nav__link">
    <span class="md-ellipsis">
      Protein
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Protein">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ContactPredictionBinary" class="md-nav__link">
    <span class="md-ellipsis">
      ContactPredictionBinary
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SspQ3" class="md-nav__link">
    <span class="md-ellipsis">
      SspQ3
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FoldPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FoldPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.LocalizationPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      LocalizationPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.MetalIonBinding" class="md-nav__link">
    <span class="md-ellipsis">
      MetalIonBinding
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SolubilityPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      SolubilityPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.AntibioticResistance" class="md-nav__link">
    <span class="md-ellipsis">
      AntibioticResistance
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.CloningClf" class="md-nav__link">
    <span class="md-ellipsis">
      CloningClf
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.MaterialProduction" class="md-nav__link">
    <span class="md-ellipsis">
      MaterialProduction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TcrPmhcAffinity" class="md-nav__link">
    <span class="md-ellipsis">
      TcrPmhcAffinity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.PeptideHlaMhcAffinity" class="md-nav__link">
    <span class="md-ellipsis">
      PeptideHlaMhcAffinity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TemperatureStability" class="md-nav__link">
    <span class="md-ellipsis">
      TemperatureStability
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FluorescencePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FluorescencePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FitnessPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FitnessPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.StabilityPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      StabilityPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.EnzymeCatalyticEfficiencyPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      EnzymeCatalyticEfficiencyPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.OptimalTemperaturePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      OptimalTemperaturePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.OptimalPhPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      OptimalPhPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DMSFitnessPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      DMSFitnessPrediction
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#structure" class="md-nav__link">
    <span class="md-ellipsis">
      Structure
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Structure">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ContactPredictionBinary" class="md-nav__link">
    <span class="md-ellipsis">
      ContactPredictionBinary
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SspQ3" class="md-nav__link">
    <span class="md-ellipsis">
      SspQ3
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FoldPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FoldPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FluorescencePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FluorescencePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DMSFitnessPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      DMSFitnessPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.StructureTokenDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      StructureTokenDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cell" class="md-nav__link">
    <span class="md-ellipsis">
      Cell
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Cell">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.CellClassificationDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      CellClassificationDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.CellClassificationLargeDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      CellClassificationLargeDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ClockDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      ClockDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.PertClassificationDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      PertClassificationDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tissue" class="md-nav__link">
    <span class="md-ellipsis">
      Tissue
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Tissue">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.CellWithNeighborDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      CellWithNeighborDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#multimodal" class="md-nav__link">
    <span class="md-ellipsis">
      Multimodal
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Multimodal">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.IsoformExpression" class="md-nav__link">
    <span class="md-ellipsis">
      IsoformExpression
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#base-classes" class="md-nav__link">
    <span class="md-ellipsis">
      Base Classes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Base Classes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DataInterface" class="md-nav__link">
    <span class="md-ellipsis">
      DataInterface
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ColumnRetrievalDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      ColumnRetrievalDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SequencesDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      SequencesDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SequenceClassificationDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      SequenceClassificationDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SequenceRegressionDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      SequenceRegressionDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TokenClassificationDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      TokenClassificationDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DiffusionDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      DiffusionDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ClassDiffusionDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      ClassDiffusionDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ConditionalDiffusionDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      ConditionalDiffusionDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.MLMDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      MLMDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tasks/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Tasks
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../adapters/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Adapters
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../callbacks/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Callbacks
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../trainer/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Trainer
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#dna" class="md-nav__link">
    <span class="md-ellipsis">
      DNA
    </span>
  </a>
  
    <nav class="md-nav" aria-label="DNA">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.NTClassification" class="md-nav__link">
    <span class="md-ellipsis">
      NTClassification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.GUEClassification" class="md-nav__link">
    <span class="md-ellipsis">
      GUEClassification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ClinvarRetrieve" class="md-nav__link">
    <span class="md-ellipsis">
      ClinvarRetrieve
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.PromoterExpressionRegression" class="md-nav__link">
    <span class="md-ellipsis">
      PromoterExpressionRegression
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.PromoterExpressionGeneration" class="md-nav__link">
    <span class="md-ellipsis">
      PromoterExpressionGeneration
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DependencyMappingDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      DependencyMappingDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#rna" class="md-nav__link">
    <span class="md-ellipsis">
      RNA
    </span>
  </a>
  
    <nav class="md-nav" aria-label="RNA">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TranslationEfficiency" class="md-nav__link">
    <span class="md-ellipsis">
      TranslationEfficiency
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ExpressionLevel" class="md-nav__link">
    <span class="md-ellipsis">
      ExpressionLevel
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TranscriptAbundance" class="md-nav__link">
    <span class="md-ellipsis">
      TranscriptAbundance
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ProteinAbundance" class="md-nav__link">
    <span class="md-ellipsis">
      ProteinAbundance
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.NcrnaFamilyClassification" class="md-nav__link">
    <span class="md-ellipsis">
      NcrnaFamilyClassification
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SpliceSitePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      SpliceSitePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ModificationSitePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      ModificationSitePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.RNAMeanRibosomeLoadDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      RNAMeanRibosomeLoadDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#protein" class="md-nav__link">
    <span class="md-ellipsis">
      Protein
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Protein">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ContactPredictionBinary" class="md-nav__link">
    <span class="md-ellipsis">
      ContactPredictionBinary
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SspQ3" class="md-nav__link">
    <span class="md-ellipsis">
      SspQ3
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FoldPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FoldPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.LocalizationPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      LocalizationPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.MetalIonBinding" class="md-nav__link">
    <span class="md-ellipsis">
      MetalIonBinding
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SolubilityPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      SolubilityPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.AntibioticResistance" class="md-nav__link">
    <span class="md-ellipsis">
      AntibioticResistance
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.CloningClf" class="md-nav__link">
    <span class="md-ellipsis">
      CloningClf
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.MaterialProduction" class="md-nav__link">
    <span class="md-ellipsis">
      MaterialProduction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TcrPmhcAffinity" class="md-nav__link">
    <span class="md-ellipsis">
      TcrPmhcAffinity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.PeptideHlaMhcAffinity" class="md-nav__link">
    <span class="md-ellipsis">
      PeptideHlaMhcAffinity
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TemperatureStability" class="md-nav__link">
    <span class="md-ellipsis">
      TemperatureStability
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FluorescencePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FluorescencePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FitnessPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FitnessPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.StabilityPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      StabilityPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.EnzymeCatalyticEfficiencyPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      EnzymeCatalyticEfficiencyPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.OptimalTemperaturePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      OptimalTemperaturePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.OptimalPhPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      OptimalPhPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DMSFitnessPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      DMSFitnessPrediction
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#structure" class="md-nav__link">
    <span class="md-ellipsis">
      Structure
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Structure">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ContactPredictionBinary" class="md-nav__link">
    <span class="md-ellipsis">
      ContactPredictionBinary
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SspQ3" class="md-nav__link">
    <span class="md-ellipsis">
      SspQ3
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FoldPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FoldPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.FluorescencePrediction" class="md-nav__link">
    <span class="md-ellipsis">
      FluorescencePrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DMSFitnessPrediction" class="md-nav__link">
    <span class="md-ellipsis">
      DMSFitnessPrediction
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.StructureTokenDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      StructureTokenDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cell" class="md-nav__link">
    <span class="md-ellipsis">
      Cell
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Cell">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.CellClassificationDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      CellClassificationDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.CellClassificationLargeDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      CellClassificationLargeDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ClockDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      ClockDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.PertClassificationDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      PertClassificationDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tissue" class="md-nav__link">
    <span class="md-ellipsis">
      Tissue
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Tissue">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.CellWithNeighborDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      CellWithNeighborDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#multimodal" class="md-nav__link">
    <span class="md-ellipsis">
      Multimodal
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Multimodal">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.IsoformExpression" class="md-nav__link">
    <span class="md-ellipsis">
      IsoformExpression
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#base-classes" class="md-nav__link">
    <span class="md-ellipsis">
      Base Classes
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Base Classes">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DataInterface" class="md-nav__link">
    <span class="md-ellipsis">
      DataInterface
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ColumnRetrievalDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      ColumnRetrievalDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SequencesDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      SequencesDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SequenceClassificationDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      SequenceClassificationDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.SequenceRegressionDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      SequenceRegressionDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.TokenClassificationDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      TokenClassificationDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.DiffusionDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      DiffusionDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ClassDiffusionDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      ClassDiffusionDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.ConditionalDiffusionDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      ConditionalDiffusionDataModule
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#modelgenerator.data.MLMDataModule" class="md-nav__link">
    <span class="md-ellipsis">
      MLMDataModule
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="data">Data</h1>
<p>Data modules specify data sources, as well as data loading and preprocessing for use with <a href="../tasks">Tasks</a>.
They provide a simple interface for swapping data sources and re-using datasets for new workflows without any code changes, enabling rapid and reproducible experimentation.
They are specified with the <code>--data</code> arguent in the CLI or in the <code>data</code> section of a configuration file.</p>
<p>Data modules can automatically load common data sources (json, tsv, txt, HuggingFace) and uncommon ones (h5ad, TileDB).
They transform, split, and sample these sources for training with <code>mgen fit</code>, evaluation with <code>mgen test/validate</code>, and inference with <code>mgen predict</code>.</p>
<p>This reference overviews the available no-code data modules.
If you would like to develop new datasets, see <a href="../../experiment_design">Experiment Design</a>.</p>
<pre><code class="language-yaml">data:
  class_path: modelgenerator.data.DMSFitnessPrediction
  init_args:
    path: genbio-ai/ProteinGYM-DMS
    train_split_files:
    - indels/B1LPA6_ECOSM_Russ_2020_indels.tsv
    train_split_name: train
    random_seed: 42
    batch_size: 32
    cv_num_folds: 5
    cv_test_fold_id: 0
    cv_enable_val_fold: true
    cv_fold_id_col: fold_id
model:
  ...
trainer:
  ...
</code></pre>
<blockquote>
<p>Note: Data modules are designed for use with a specific task, indicated in the class name.</p>
</blockquote>
<h2 id="dna">DNA</h2>


<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.NTClassification" class="doc doc-heading">
            <code>modelgenerator.data.NTClassification</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Nucleotide Transformer benchmarks from InstaDeep.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.01.11.523679v3">The Nucleotide Transformer: Building and Evaluating Robust Foundation Models for Human Genomics</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/InstaDeepAI/nucleotide_transformer_downstream_tasks">InstaDeepAI/nucleotide_transformer_downstream_tasks</a></li>
<li>Configs:<ul>
<li><code>promoter_all</code></li>
<li><code>promoter_tata</code></li>
<li><code>promoter_no_tata</code></li>
<li><code>enhancers</code></li>
<li><code>enhancers_types</code></li>
<li><code>splice_sites_all</code></li>
<li><code>splice_sites_acceptor</code></li>
<li><code>splice_sites_donor</code></li>
<li><code>H3</code></li>
<li><code>H4</code></li>
<li><code>H3K9ac</code></li>
<li><code>H3K14ac</code></li>
<li><code>H4ac</code></li>
<li><code>H3K4me1</code></li>
<li><code>H3K4me2</code></li>
<li><code>H3K4me3</code></li>
<li><code>H3K36me3</code></li>
<li><code>H3K79me3</code></li>
</ul>
</li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;InstaDeepAI/nucleotide_transformer_downstream_tasks&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;enhancers&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequence&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;sequence&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.GUEClassification" class="doc doc-heading">
            <code>modelgenerator.data.GUEClassification</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Genome Understanding Evaluation benchmarks for DNABERT-2 from the Liu Lab at Northwestern.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://arxiv.org/abs/2306.15006">DNABERT-2: Efficient Foundation Model and Benchmark For Multi-Species Genome</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/leannmlindsey/GUE">leannmlindsey/GUE</a></li>
<li>Configs:<ul>
<li><code>emp_H3</code></li>
<li><code>emp_H3K14ac</code></li>
<li><code>emp_H3K36me3</code></li>
<li><code>emp_H3K4me1</code></li>
<li><code>emp_H3K4me2</code></li>
<li><code>emp_H3K4me3</code></li>
<li><code>emp_H3K79me3</code></li>
<li><code>emp_H3K9ac</code></li>
<li><code>emp_H4</code></li>
<li><code>emp_H4ac</code></li>
<li><code>human_tf_0</code></li>
<li><code>human_tf_1</code></li>
<li><code>human_tf_2</code></li>
<li><code>human_tf_3</code></li>
<li><code>human_tf_4</code></li>
<li><code>mouse_0</code></li>
<li><code>mouse_1</code></li>
<li><code>mouse_2</code></li>
<li><code>mouse_3</code></li>
<li><code>mouse_4</code></li>
<li><code>prom_300_all</code></li>
<li><code>prom_300_notata</code></li>
<li><code>prom_300_tata</code></li>
<li><code>prom_core_all</code></li>
<li><code>prom_core_notata</code></li>
<li><code>prom_core_tata</code></li>
<li><code>splice_reconstructed</code></li>
<li><code>virus_covid</code></li>
<li><code>virus_species_40</code></li>
<li><code>fungi_species_20</code></li>
<li><code>EPI_K562</code></li>
<li><code>EPI_HeLa-S3</code></li>
<li><code>EPI_NHEK</code></li>
<li><code>EPI_IMR90</code></li>
<li><code>EPI_HUVEC</code></li>
</ul>
</li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;leannmlindsey/GUE&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;emp_H3&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequence&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;sequence&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ClinvarRetrieve" class="doc doc-heading">
            <code>modelgenerator.data.ClinvarRetrieve</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><span title="modelgenerator.data.data.ZeroshotClassificationRetrieveDataModule">ZeroshotClassificationRetrieveDataModule</span></code></p>


        <p>ClinVar dataset for genomic variant effect prediction.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.01.11.523679v3">The Nucleotide Transformer: Building and Evaluating Robust Foundation Models for Human Genomics</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/Clinvar">genbio-ai/Clinvar</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>List</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>[&#39;ClinVar_Processed.tsv&#39;]</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>reference_file</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The file path to the reference file for retrieving sequences</p>
              </div>
            </td>
            <td>
                  <code>&#39;hg38.ml.fa&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>method</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>method mode to compute metrics</p>
              </div>
            </td>
            <td>
                  <code>&#39;Distance&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>window</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of token taken on either side of the mutation site. The processed sequence length is <code>2 * window + 1</code></p>
              </div>
            </td>
            <td>
                  <code>512</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>index_cols</code>
            </td>
            <td>
                  <code>List</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The list of the column name containing the index for sequence retrieval.</p>
              </div>
            </td>
            <td>
                  <code>[&#39;chrom&#39;, &#39;start&#39;, &#39;end&#39;, &#39;ref&#39;, &#39;mutate&#39;]</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the labels. Defaults to "label".</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.PromoterExpressionRegression" class="doc doc-heading">
            <code>modelgenerator.data.PromoterExpressionRegression</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Gene expression prediction from promoter sequences from the Regev Lab at the Broad Institute.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.nature.com/articles/s41587-019-0315-8">Deciphering eukaryotic gene-regulatory logic with 100 million random promoters</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/100M-random-promoters">genbio-ai/100M-random-promoters</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/100M-random-promoters&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequence&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;sequence&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.PromoterExpressionGeneration" class="doc doc-heading">
            <code>modelgenerator.data.PromoterExpressionGeneration</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.ConditionalDiffusionDataModule&lt;/code&gt;" href="#modelgenerator.data.ConditionalDiffusionDataModule">ConditionalDiffusionDataModule</a></code></p>


        <p>Promoter generation from gene expression data from the Regev Lab at the Broad Institute.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.nature.com/articles/s41587-019-0315-8">Deciphering eukaryotic gene-regulatory logic with 100 million random promoters</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/100M-random-promoters">genbio-ai/100M-random-promoters</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/100M-random-promoters&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequence&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;sequence&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>timesteps_per_sample</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of timesteps per sample.</p>
              </div>
            </td>
            <td>
                  <code>10</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>randomize_targets</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to randomize the target sequences for each timestep (experimental efficiency boost).</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.DependencyMappingDataModule" class="doc doc-heading">
            <code>modelgenerator.data.DependencyMappingDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequencesDataModule&lt;/code&gt;" href="#modelgenerator.data.SequencesDataModule">SequencesDataModule</a></code></p>


        <p>Data module for doing dependency mapping via in silico mutagenesis on a dataset of sequences.
Only uses the test set.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes a single sequence under key 'sequences' and optionally an 'ids' to track outputs.</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>vocab_file</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The path to the file with the vocabulary to mutate.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the sequences. Defaults to "sequence".</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequence&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>id_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the ids. Defaults to "id".</p>
              </div>
            </td>
            <td>
                  <code>&#39;id&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div><h2 id="rna">RNA</h2>


<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.TranslationEfficiency" class="doc doc-heading">
            <code>modelgenerator.data.TranslationEfficiency</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Translation efficiency prediction benchmarks from the Wang Lab at Princeton.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.nature.com/articles/s42256-024-00823-9">A 5 UTR language model for decoding untranslated regions of mRNA and function predictions</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/rna-downstream-tasks">genbio-ai/rna-downstream-tasks</a></li>
<li>Configs:<ul>
<li><code>translation_efficiency_Muscle</code></li>
<li><code>translation_efficiency_HEK</code></li>
<li><code>translation_efficiency_pc3</code></li>
</ul>
</li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/rna-downstream-tasks&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;translation_efficiency_Muscle&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>10</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>&#39;fold_id&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ExpressionLevel" class="doc doc-heading">
            <code>modelgenerator.data.ExpressionLevel</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Expression level prediction benchmarks from the Wang Lab at Princeton.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.nature.com/articles/s42256-024-00823-9">A 5 UTR language model for decoding untranslated regions of mRNA and function predictions</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/rna-downstream-tasks">genbio-ai/rna-downstream-tasks</a></li>
<li>Configs:<ul>
<li><code>expression_Muscle</code></li>
<li><code>expression_HEK</code></li>
<li><code>expression_pc3</code></li>
</ul>
</li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/rna-downstream-tasks&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;expression_Muscle&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>10</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>&#39;fold_id&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.TranscriptAbundance" class="doc doc-heading">
            <code>modelgenerator.data.TranscriptAbundance</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Transcript abundance prediction benchmarks from the Wang Lab at Princeton.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.nature.com/articles/s42256-024-00823-9">A 5 UTR language model for decoding untranslated regions of mRNA and function predictions</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/rna-downstream-tasks">genbio-ai/rna-downstream-tasks</a></li>
<li>Configs:<ul>
<li><code>transcript_abundance_athaliana</code></li>
<li><code>transcript_abundance_dmelanogaster</code></li>
<li><code>transcript_abundance_ecoli</code></li>
<li><code>transcript_abundance_hsapiens</code></li>
<li><code>transcript_abundance_hvolcanii</code></li>
<li><code>transcript_abundance_ppastoris</code></li>
<li><code>transcript_abundance_scerevisiae</code></li>
</ul>
</li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/rna-downstream-tasks&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;transcript_abundance_athaliana&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>5</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>&#39;fold_id&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ProteinAbundance" class="doc doc-heading">
            <code>modelgenerator.data.ProteinAbundance</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Protein abundance prediction benchmarks from the Wang Lab at Princeton.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.nature.com/articles/s42256-024-00823-9">A 5 UTR language model for decoding untranslated regions of mRNA and function predictions</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/rna-downstream-tasks">genbio-ai/rna-downstream-tasks</a></li>
<li>Configs:<ul>
<li><code>protein_abundance_athaliana</code></li>
<li><code>protein_abundance_dmelanogaster</code></li>
<li><code>protein_abundance_ecoli</code></li>
<li><code>protein_abundance_hsapiens</code></li>
<li><code>protein_abundance_scerevisiae</code></li>
</ul>
</li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/rna-downstream-tasks&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;protein_abundance_athaliana&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>5</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>&#39;fold_id&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.NcrnaFamilyClassification" class="doc doc-heading">
            <code>modelgenerator.data.NcrnaFamilyClassification</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Non-coding RNA family classification benchmarks from DPTechnology.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.11.548588v1">UNI-RNA: UNIVERSAL PRE-TRAINED MODELS REVOLUTIONIZE RNA RESEARCH</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/rna-downstream-tasks">genbio-ai/rna-downstream-tasks</a></li>
<li>Configs:<ul>
<li><code>ncrna_family_bnoise0</code></li>
<li><code>ncrna_family_bnoise200</code></li>
</ul>
</li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/rna-downstream-tasks&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;ncrna_family_bnoise0&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;validation&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.SpliceSitePrediction" class="doc doc-heading">
            <code>modelgenerator.data.SpliceSitePrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Splice site prediction benchmarks from the Thompson Lab at University of Strasbourg.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04471-3">Spliceator: multi-species splice site prediction using convolutional neural networks</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/rna-downstream-tasks">genbio-ai/rna-downstream-tasks</a></li>
<li>Configs:<ul>
<li><code>splice_site_acceptor</code></li>
<li><code>splice_site_donor</code></li>
</ul>
</li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/rna-downstream-tasks&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;splice_site_acceptor&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;validation&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test_danio&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>16</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ModificationSitePrediction" class="doc doc-heading">
            <code>modelgenerator.data.ModificationSitePrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Modification site prediction benchmarks from the Meng Lab at the University of Liverpool.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.nature.com/articles/s41467-021-24313-3">Attention-based multi-label neural networks for integrated prediction and interpretation of twelve widely occurring RNA modifications</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/rna-downstream-tasks">genbio-ai/rna-downstream-tasks</a></li>
<li>Configs:<ul>
<li><code>modification_site</code></li>
</ul>
</li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/rna-downstream-tasks&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;modification_site&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>List</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>[&#39;labels_0&#39;, &#39;labels_1&#39;, &#39;labels_2&#39;, &#39;labels_3&#39;, &#39;labels_4&#39;, &#39;labels_5&#39;, &#39;labels_6&#39;, &#39;labels_7&#39;, &#39;labels_8&#39;, &#39;labels_9&#39;, &#39;labels_10&#39;, &#39;labels_11&#39;]</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;validation&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.RNAMeanRibosomeLoadDataModule" class="doc doc-heading">
            <code>modelgenerator.data.RNAMeanRibosomeLoadDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Data module for the mean ribosome load dataset.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.nature.com/articles/s41587-019-0164-5">Human 5' UTR design and variant effect prediction from a massively parallel translation assay</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/rna-downstream-tasks">genbio-ai/rna-downstream-tasks</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/rna-downstream-tasks&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>&#39;mean_ribosome_load&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;validation&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;utr&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;rl&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;utr&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div><h2 id="protein">Protein</h2>


<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ContactPredictionBinary" class="doc doc-heading">
            <code>modelgenerator.data.ContactPredictionBinary</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.TokenClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.TokenClassificationDataModule">TokenClassificationDataModule</a></code></p>


        <p>Protein contact prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/contact_prediction_binary">proteinglm/contact_prediction_binary</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/contact_prediction_binary&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pairwise</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the labels are pairwise.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_cols</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Additional columns to include in the dataset.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_length</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The maximum length of the sequences.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>truncate_extra_cols</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to truncate the extra columns to the maximum length.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.SspQ3" class="doc doc-heading">
            <code>modelgenerator.data.SspQ3</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.TokenClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.TokenClassificationDataModule">TokenClassificationDataModule</a></code></p>


        <p>Protein secondary structure prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/ssp_q3">proteinglm/ssp_q3</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/ssp_q3&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pairwise</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the labels are pairwise.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_cols</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Additional columns to include in the dataset.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_length</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The maximum length of the sequences.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>truncate_extra_cols</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to truncate the extra columns to the maximum length.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.FoldPrediction" class="doc doc-heading">
            <code>modelgenerator.data.FoldPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Protein fold prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/fold_prediction">proteinglm/fold_prediction</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/fold_prediction&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.LocalizationPrediction" class="doc doc-heading">
            <code>modelgenerator.data.LocalizationPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Protein localization prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/localization_prediction">proteinglm/localization_prediction</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/localization_prediction&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.MetalIonBinding" class="doc doc-heading">
            <code>modelgenerator.data.MetalIonBinding</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Metal ion binding prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/metal_ion_binding">proteinglm/metal_ion_binding</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/metal_ion_binding&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.SolubilityPrediction" class="doc doc-heading">
            <code>modelgenerator.data.SolubilityPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Protein solubility prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/solubility_prediction">proteinglm/solubility_prediction</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/solubility_prediction&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.AntibioticResistance" class="doc doc-heading">
            <code>modelgenerator.data.AntibioticResistance</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Antibiotic resistance prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/antibiotic_resistance">proteinglm/antibiotic_resistance</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/antibiotic_resistance&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.CloningClf" class="doc doc-heading">
            <code>modelgenerator.data.CloningClf</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Cloning classification prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/cloning_clf">proteinglm/cloning_clf</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/cloning_clf&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.MaterialProduction" class="doc doc-heading">
            <code>modelgenerator.data.MaterialProduction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Material production prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/material_production">proteinglm/material_production</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/material_production&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.TcrPmhcAffinity" class="doc doc-heading">
            <code>modelgenerator.data.TcrPmhcAffinity</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>TCR-pMHC affinity prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/tcr_pmhc_affinity">proteinglm/tcr_pmhc_affinity</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/tcr_pmhc_affinity&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.PeptideHlaMhcAffinity" class="doc doc-heading">
            <code>modelgenerator.data.PeptideHlaMhcAffinity</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Peptide-HLA-MHC affinity prediction benchmarks from BioMap.
Note:
    - Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a>
    - Data Card: <a href="https://huggingface.co/datasets/proteinglm/peptide_HLA_MHC_affinity">proteinglm/peptide_HLA_MHC_affinity</a></p>


<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/peptide_HLA_MHC_affinity&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.TemperatureStability" class="doc doc-heading">
            <code>modelgenerator.data.TemperatureStability</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Temperature stability prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/temperature_stability">proteinglm/temperature_stability</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/temperature_stability&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.FluorescencePrediction" class="doc doc-heading">
            <code>modelgenerator.data.FluorescencePrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Fluorescence prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/fluorescence_prediction">proteinglm/fluorescence_prediction</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/fluorescence_prediction&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.FitnessPrediction" class="doc doc-heading">
            <code>modelgenerator.data.FitnessPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Fitness prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/fitness_prediction">proteinglm/fitness_prediction</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/fitness_prediction&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.StabilityPrediction" class="doc doc-heading">
            <code>modelgenerator.data.StabilityPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Stability prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/stability_prediction">proteinglm/stability_prediction</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/stability_prediction&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.EnzymeCatalyticEfficiencyPrediction" class="doc doc-heading">
            <code>modelgenerator.data.EnzymeCatalyticEfficiencyPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Enzyme catalytic efficiency prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/enzyme_catalytic_efficiency">proteinglm/enzyme_catalytic_efficiency</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/enzyme_catalytic_efficiency&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.OptimalTemperaturePrediction" class="doc doc-heading">
            <code>modelgenerator.data.OptimalTemperaturePrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Optimal temperature prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/optimal_temperature">proteinglm/optimal_temperature</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/optimal_temperature&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.OptimalPhPrediction" class="doc doc-heading">
            <code>modelgenerator.data.OptimalPhPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Optimal pH prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/optimal_ph">proteinglm/optimal_ph</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/optimal_ph&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.DMSFitnessPrediction" class="doc doc-heading">
            <code>modelgenerator.data.DMSFitnessPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Deep mutational scanning (DMS) fitness prediction benchmarks from the Gal Lab at Oxford and the Marks Lab at Harvard.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://proteingym.org/">ProteinGym: Large-Scale Benchmarks for Protein Fitness Prediction and Design</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/ProteinGYM-DMS">genbio-ai/ProteinGYM-DMS</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/ProteinGYM-DMS&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>list</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>[&#39;indels/B1LPA6_ECOSM_Russ_2020_indels.tsv&#39;]</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>5</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>&#39;fold_id&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>-1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div><h2 id="structure">Structure</h2>


<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ContactPredictionBinary" class="doc doc-heading">
            <code>modelgenerator.data.ContactPredictionBinary</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.TokenClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.TokenClassificationDataModule">TokenClassificationDataModule</a></code></p>


        <p>Protein contact prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/contact_prediction_binary">proteinglm/contact_prediction_binary</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/contact_prediction_binary&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pairwise</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the labels are pairwise.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_cols</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Additional columns to include in the dataset.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_length</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The maximum length of the sequences.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>truncate_extra_cols</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to truncate the extra columns to the maximum length.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.SspQ3" class="doc doc-heading">
            <code>modelgenerator.data.SspQ3</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.TokenClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.TokenClassificationDataModule">TokenClassificationDataModule</a></code></p>


        <p>Protein secondary structure prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/ssp_q3">proteinglm/ssp_q3</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/ssp_q3&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pairwise</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the labels are pairwise.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_cols</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Additional columns to include in the dataset.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_length</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The maximum length of the sequences.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>truncate_extra_cols</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to truncate the extra columns to the maximum length.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.FoldPrediction" class="doc doc-heading">
            <code>modelgenerator.data.FoldPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Protein fold prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/fold_prediction">proteinglm/fold_prediction</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/fold_prediction&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.FluorescencePrediction" class="doc doc-heading">
            <code>modelgenerator.data.FluorescencePrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Fluorescence prediction benchmarks from BioMap.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://www.biorxiv.org/content/10.1101/2023.07.05.547496v5">xTrimoPGLM: Unified 100B-Scale Pre-trained Transformer for Deciphering the Language of Protein</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/proteinglm/fluorescence_prediction">proteinglm/fluorescence_prediction</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;proteinglm/fluorescence_prediction&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;seq&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;label&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;seq&#39;: &#39;sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.DMSFitnessPrediction" class="doc doc-heading">
            <code>modelgenerator.data.DMSFitnessPrediction</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Deep mutational scanning (DMS) fitness prediction benchmarks from the Gal Lab at Oxford and the Marks Lab at Harvard.</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://proteingym.org/">ProteinGym: Large-Scale Benchmarks for Protein Fitness Prediction and Design</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/ProteinGYM-DMS">genbio-ai/ProteinGYM-DMS</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/ProteinGYM-DMS&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>list</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>[&#39;indels/B1LPA6_ECOSM_Russ_2020_indels.tsv&#39;]</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>5</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>&#39;fold_id&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>-1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_context_length</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Maximum context length for the input sequences.</p>
              </div>
            </td>
            <td>
                  <code>12800</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>msa_random_seed</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Random seed for MSA generation.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>is_rag_dataset</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the dataset is a RAG dataset for AIDO.Protein-RAG.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.StructureTokenDataModule" class="doc doc-heading">
            <code>modelgenerator.data.StructureTokenDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code>, <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.HFDatasetLoaderMixin&lt;/code&gt;" href="../../experiment_design/data/#modelgenerator.data.HFDatasetLoaderMixin">HFDatasetLoaderMixin</a></code></p>


        <p>Test only data module for structure token predictors.</p>
<p>This data module is specifically designed for handling datasets uses amino acid sequences as input
and structure tokens as labels.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>This module only supports testing and ignores training and validation splits.
It assumes test split files contain sequences and optionally their structural token labels.
If structural token labels are not provided, dummy labels are created.</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div><h2 id="cell">Cell</h2>


<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.CellClassificationDataModule" class="doc doc-heading">
            <code>modelgenerator.data.CellClassificationDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code></p>


        <p>Data module for cell classification.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes a feature vector (one of the rows in <adata.X>) and a single class label (one of the columns in <adata.obs>)</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>backbone_class_path</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Class path of the backbone model.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>filter_columns</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The columns of <obs> we want to use. Defaults to None, in which case all columns are used.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_columns</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>New name of columns. Defaults to None, in which case columns are not renamed. Does nothing if filter_colums is None.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.CellClassificationLargeDataModule" class="doc doc-heading">
            <code>modelgenerator.data.CellClassificationLargeDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code></p>


        <p>Data module for cell classification. This class handles large dataset and is implemented based on TileDB.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes a feature vector (one of the rows in <adata.X>) and a single class label (one of the columns in <adata.obs>)</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the TileDB dataset folder</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_subfolder</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Subfolder name for the training split.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_subfolder</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Subfolder name for the validation split.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_subfolder</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Subfolder name for the test split.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>backbone_class_path</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Class path of the backbone model.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>layer_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Name of the layer in the TileDB dataset.</p>
              </div>
            </td>
            <td>
                  <code>&#39;data&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>obs_column_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Name of the column in <obs> to use as the label.</p>
              </div>
            </td>
            <td>
                  <code>&#39;cell_type&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>measurement_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Name of the measurement in the TileDB dataset.</p>
              </div>
            </td>
            <td>
                  <code>&#39;RNA&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>axis_query_value_filter</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Optional filter for the axis query.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>prefetch_factor</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Number of batches to prefetch.</p>
              </div>
            </td>
            <td>
                  <code>16</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ClockDataModule" class="doc doc-heading">
            <code>modelgenerator.data.ClockDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code></p>


        <p>Data module for transcriptomic clock tasks.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes a feature vector (one of the rows in <adata.X>) and a single scalar corresponding to donor age (one of the columns in <adata.obs>)</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>split_column</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column of <obs> that defines the split assignments.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>label_scaling</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The type of label scaling to apply.</p>
              </div>
            </td>
            <td>
                  <code>&#39;z_scaling&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>backbone_class_path</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Class path of the backbone model.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>filter_columns</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The columns of <obs> we want to use. Defaults to None, in which case all columns are used.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_columns</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>New name of columns. Defaults to None, in which case columns are not renamed. Does nothing if filter_colums is None.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.PertClassificationDataModule" class="doc doc-heading">
            <code>modelgenerator.data.PertClassificationDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code></p>


        <p>Data module for perturbation classification.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes a feature vector (one of the rows in <adata.X>) and a single class label (one of the columns in <adata.obs>)</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pert_column</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Column of <obs> containing perturbation labels.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cell_line_column</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Column of <obs> containing cell line labels.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cell_line</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Name of cell line to consider.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>split_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Seed for train/val/test splits.</p>
              </div>
            </td>
            <td>
                  <code>1234</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_frac</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Fraction of examples to assign to train set.</p>
              </div>
            </td>
            <td>
                  <code>0.7</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>val_frac</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Fraction of examples to assign to val set.</p>
              </div>
            </td>
            <td>
                  <code>0.15</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_frac</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Fraction of examples to assign to test set.</p>
              </div>
            </td>
            <td>
                  <code>0.15</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>backbone_class_path</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Class path of the backbone model.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>filter_columns</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The columns of <obs> we want to use. Defaults to None, in which case all columns are used.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_columns</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>New name of columns. Defaults to None, in which case columns are not renamed. Does nothing if filter_colums is None.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div><h2 id="tissue">Tissue</h2>


<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.CellWithNeighborDataModule" class="doc doc-heading">
            <code>modelgenerator.data.CellWithNeighborDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code></p>


        <p>Data module for cell classification with neighbors for AIDO.Tissue.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes a feature vector (one of the rows in <adata.X>) and a single class label (one of the columns in <adata.obs>)
The feature vector is concatenated with the feature vectors of its neighbors.</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>filter_columns</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The columns of <obs> we want to use. Defaults to None, in which case all columns are used.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_columns</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Optional list of columns to rename.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>use_random_neighbor</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use random neighbors.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>copy_center_as_neighbor</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to copy center as a neighbor.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>neighbor_num</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Number of neighbors to consider.</p>
              </div>
            </td>
            <td>
                  <code>10</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique identifier.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div><h2 id="multimodal">Multimodal</h2>


<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.IsoformExpression" class="doc doc-heading">
            <code>modelgenerator.data.IsoformExpression</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Isoform expression prediction benchmarks from the</p>


<details class="note" open>
  <summary>Note</summary>
  <ul>
<li>Manuscript: <a href="https://arxiv.org/abs/2406.14150">Multi-modal Transfer Learning between Biological Foundation Models</a></li>
<li>Data Card: <a href="https://huggingface.co/datasets/genbio-ai/transcript_isoform_expression_prediction">genbio-ai/transcript_isoform_expression_prediction</a></li>
</ul>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                  <code>&#39;genbio-ai/transcript_isoform_expression_prediction&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>[&#39;dna_seq&#39;, &#39;rna_seq&#39;, &#39;protein_seq&#39;]</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>{&#39;dna_seq&#39;: &#39;dna_sequences&#39;, &#39;rna_seq&#39;: &#39;rna_sequences&#39;, &#39;protein_seq&#39;: &#39;protein_sequences&#39;}</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;valid&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train_*.tsv&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test.tsv&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>&#39;validation.tsv&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div><h2 id="base-classes">Base Classes</h2>


<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.DataInterface" class="doc doc-heading">
            <code>modelgenerator.data.DataInterface</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><span title="lightning.pytorch.LightningDataModule">LightningDataModule</span></code>, <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.KFoldMixin&lt;/code&gt;" href="../../experiment_design/data/#modelgenerator.data.KFoldMixin">KFoldMixin</a></code></p>


        <p>Base class for all data modules in this project. Handles the boilerplate of setting up data loaders.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Subclasses must implement the setup method.
All datasets should return a dictionary of data items.
To use HF loading, add the HFDatasetLoaderMixin.
For any task-specific behaviors, implement transformations using <code>torch.utils.data.Dataset</code> objects.
See <a href="./#modelgenerator.data.MLMDataModule">MLM</a> for an example.</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ColumnRetrievalDataModule" class="doc doc-heading">
            <code>modelgenerator.data.ColumnRetrievalDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code>, <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.HFDatasetLoaderMixin&lt;/code&gt;" href="../../experiment_design/data/#modelgenerator.data.HFDatasetLoaderMixin">HFDatasetLoaderMixin</a></code></p>


        <p>Simple data module for retrieving and renaming columns from a dataset.</p>


<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>in_cols</code>
            </td>
            <td>
                  <code>List</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the columns to retrieve.</p>
              </div>
            </td>
            <td>
                  <code>[]</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>out_cols</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the columns to use as the alias for the retrieved columns.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.SequencesDataModule" class="doc doc-heading">
            <code>modelgenerator.data.SequencesDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code>, <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.HFDatasetLoaderMixin&lt;/code&gt;" href="../../experiment_design/data/#modelgenerator.data.HFDatasetLoaderMixin">HFDatasetLoaderMixin</a></code></p>


        <p>Data module for loading a simple dataset of sequences.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes a single sequence under key 'sequences' and optionally an 'id' to track outputs.</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequence&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>id_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the ids.</p>
              </div>
            </td>
            <td>
                  <code>&#39;id&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.SequenceClassificationDataModule" class="doc doc-heading">
            <code>modelgenerator.data.SequenceClassificationDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><span title="modelgenerator.data.data.ClassificationDataModule">ClassificationDataModule</span></code>, <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.HFDatasetLoaderMixin&lt;/code&gt;" href="../../experiment_design/data/#modelgenerator.data.HFDatasetLoaderMixin">HFDatasetLoaderMixin</a></code></p>


        <p>Data module for Hugging Face sequence classification datasets.</p>


<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.SequenceRegressionDataModule" class="doc doc-heading">
            <code>modelgenerator.data.SequenceRegressionDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><span title="modelgenerator.data.data.RegressionDataModule">RegressionDataModule</span></code>, <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.HFDatasetLoaderMixin&lt;/code&gt;" href="../../experiment_design/data/#modelgenerator.data.HFDatasetLoaderMixin">HFDatasetLoaderMixin</a></code></p>


        <p>Data module for sequence regression datasets.</p>


<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.TokenClassificationDataModule" class="doc doc-heading">
            <code>modelgenerator.data.TokenClassificationDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code>, <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.HFDatasetLoaderMixin&lt;/code&gt;" href="../../experiment_design/data/#modelgenerator.data.HFDatasetLoaderMixin">HFDatasetLoaderMixin</a></code></p>


        <p>Data module for Hugging Face token classification datasets.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes a single sequence under key 'sequences' and a single class sequence under key 'labels'</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_cols</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Additional columns to include in the dataset.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>max_length</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The maximum length of the sequences.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>truncate_extra_cols</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to truncate the extra columns to the maximum length.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pairwise</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether the labels are pairwise.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.DiffusionDataModule" class="doc doc-heading">
            <code>modelgenerator.data.DiffusionDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.DataInterface&lt;/code&gt;" href="#modelgenerator.data.DataInterface">DataInterface</a></code>, <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.base.HFDatasetLoaderMixin&lt;/code&gt;" href="../../experiment_design/data/#modelgenerator.data.HFDatasetLoaderMixin">HFDatasetLoaderMixin</a></code></p>


        <p>Data module for datasets with discrete diffusion-based noising and loss weights from <a href="https://arxiv.org/abs/2406.07524">MDLM</a>.</p>


<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column with the data to train on.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>timesteps_per_sample</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of timesteps per sample.</p>
              </div>
            </td>
            <td>
                  <code>10</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>randomize_targets</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to randomize the target sequences for each timestep (experimental efficiency boost).</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>


<details class="notes" open>
  <summary>Notes</summary>
  <p>Each sample includes timesteps_per_sample sequences at different noise levels
Each sample's target sequences are under 'target_sequences', the input sequences are under 'sequences', and posterior weights are under 'posterior_weights'</p>
</details>








  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ClassDiffusionDataModule" class="doc doc-heading">
            <code>modelgenerator.data.ClassDiffusionDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Data module for conditional (or class-filtered) diffusion, and applying discrete diffusion noising.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes timesteps_per_sample sequences at different noise levels
Each sample's target sequences are under 'target_seqs', the input sequences are under 'input_seqs', and posterior weights are under 'posterior_weights'</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>timesteps_per_sample</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of timesteps per sample.</p>
              </div>
            </td>
            <td>
                  <code>10</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>randomize_targets</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to randomize the target sequences for each timestep (experimental efficiency boost).</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.ConditionalDiffusionDataModule" class="doc doc-heading">
            <code>modelgenerator.data.ConditionalDiffusionDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceRegressionDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceRegressionDataModule">SequenceRegressionDataModule</a></code></p>


        <p>Data module for conditional diffusion with a continuous condition, and applying discrete diffusion noising.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes timesteps_per_sample sequences at different noise levels
Each sample's target sequences are under 'target_seqs', the input sequences are under 'input_seqs', and posterior weights are under 'posterior_weights'</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of column(s) containing the sequences.</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of columns(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>normalize</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to normalize the labels.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>timesteps_per_sample</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of timesteps per sample.</p>
              </div>
            </td>
            <td>
                  <code>10</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>randomize_targets</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to randomize the target sequences for each timestep (experimental efficiency boost).</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>

<div class="doc doc-object doc-class">



<h3 id="modelgenerator.data.MLMDataModule" class="doc doc-heading">
            <code>modelgenerator.data.MLMDataModule</code>


</h3>


    <div class="doc doc-contents first">
            <p class="doc doc-class-bases">
              Bases: <code><a class="autorefs autorefs-internal" title="&lt;code&gt;modelgenerator.data.data.SequenceClassificationDataModule&lt;/code&gt;" href="#modelgenerator.data.SequenceClassificationDataModule">SequenceClassificationDataModule</a></code></p>


        <p>Data module for continuing pretraining on a masked language modeling task.</p>


<details class="note" open>
  <summary>Note</summary>
  <p>Each sample includes a single sequence under key 'sequences' and a single target sequence under key 'target_sequences'</p>
</details>

<p><span class="doc-section-title">Parameters:</span></p>
    <table>
      <thead>
        <tr>
          <th>Name</th>
          <th>Type</th>
          <th>Description</th>
          <th>Default</th>
        </tr>
      </thead>
      <tbody>
          <tr class="doc-section-item">
            <td>
                <code>path</code>
            </td>
            <td>
                  <code>str</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Path to the dataset, can be (1) a local path to a data folder or (2) a Huggingface dataset identifier.</p>
              </div>
            </td>
            <td>
                <em>required</em>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>config_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the HF dataset configuration.
Affects how the dataset is loaded.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>x_col</code>
            </td>
            <td>
                  <code><span title="str">str</span></code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column containing the sequences. Defaults to "sequences".</p>
              </div>
            </td>
            <td>
                  <code>&#39;sequences&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>y_col</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the column(s) containing the labels.</p>
              </div>
            </td>
            <td>
                  <code>&#39;labels&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>masking_rate</code>
            </td>
            <td>
                  <code><span title="float">float</span></code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The masking rate. Defaults to 0.15.</p>
              </div>
            </td>
            <td>
                  <code>0.15</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>rename_cols</code>
            </td>
            <td>
                  <code>dict[str, str] | None</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>A dictionary mapping the original column names to the new column names.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>class_filter</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Filter the dataset to only include samples with the specified class(es).</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>generate_uid</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to generate a unique ID for each sample.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the training split.</p>
              </div>
            </td>
            <td>
                  <code>&#39;train&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the test split. Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>&#39;test&#39;</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_name</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The name of the validation split.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>train_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "train" from these files.
Not used unless referenced by the name "train" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "test" from these files.
Not used unless referenced by the name "test" in one of the split_name arguments.
Also used for <code>mgen predict</code>.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_files</code>
            </td>
            <td>
                  <code>Union</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Create a split called "valid" from these files.
Not used unless referenced by the name "valid" in one of the split_name arguments.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>test_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the test split.
If test_split_name is None, creates a test split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.2</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>valid_split_size</code>
            </td>
            <td>
                  <code>float</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The size of the validation split.
If valid_split_name is None, creates a validation split of this size from the training split.</p>
              </div>
            </td>
            <td>
                  <code>0.1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>random_seed</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The random seed to use for splitting the data.</p>
              </div>
            </td>
            <td>
                  <code>42</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>extra_reader_kwargs</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Extra kwargs for dataset readers.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>batch_size</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The batch size.</p>
              </div>
            </td>
            <td>
                  <code>128</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>shuffle</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to shuffle the data.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>sampler</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The sampler to use.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>num_workers</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of workers to use for data loading.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>collate_fn</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The function to use for collating data.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>pin_memory</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to pin memory.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>persistent_workers</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to use persistent workers.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_num_folds</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The number of cross-validation folds, disables cv when &lt;= 1.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_test_fold_id</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The fold id to use for cross-validation evaluation.</p>
              </div>
            </td>
            <td>
                  <code>0</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_enable_val_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Whether to enable a validation fold.</p>
              </div>
            </td>
            <td>
                  <code>True</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_replace_val_fold_as_test_fold</code>
            </td>
            <td>
                  <code>bool</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>Replace validation fold with test fold. Only used when cv_enable_val_fold is False.</p>
              </div>
            </td>
            <td>
                  <code>False</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_fold_id_col</code>
            </td>
            <td>
                  <code>Optional</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The column name containing the fold id from a pre-split dataset. Setting to None to enable automatic splitting.</p>
              </div>
            </td>
            <td>
                  <code>None</code>
            </td>
          </tr>
          <tr class="doc-section-item">
            <td>
                <code>cv_val_offset</code>
            </td>
            <td>
                  <code>int</code>
            </td>
            <td>
              <div class="doc-md-description">
                <p>The offset applied to cv_test_fold_id to determine val_fold_id.</p>
              </div>
            </td>
            <td>
                  <code>1</code>
            </td>
          </tr>
      </tbody>
    </table>









  <div class="doc doc-children">











  </div>

    </div>

</div>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"/></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../backbones/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Backbones">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Backbones
              </div>
            </div>
          </a>
        
        
          
          <a href="../tasks/" class="md-footer__link md-footer__link--next" aria-label="Next: Tasks">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                Tasks
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      All rights reserved &copy; 2024-Present, GenBio AI
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
      <div class="md-progress" data-md-component="progress" role="progressbar"></div>
    
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["announce.dismiss", "content.code.annotate", "content.code.copy", "content.code.select", "content.tooltips", "content.tabs.link", "content.tooltips", "header.autohide", "navigation.expand", "navigation.footer", "navigation.indexes", "navigation.instant", "navigation.instant.prefetch", "navigation.instant.progress", "navigation.prune", "navigation.sections", "navigation.tabs", "navigation.tabs.sticky", "navigation.top", "navigation.tracking", "search.highlight", "search.share", "search.suggest", "toc.follow"], "search": "../../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.83f73b43.min.js"></script>
      
    
  </body>
</html>